pkg/asset/manifests/azure/cluster.go

```diff
@@ -63,6 +64,18 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 		},
 	}
 
+	// Setting ID on the Subnet disables natgw creation. See:
+	// https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/21479a9a4c640b43e0bef028487c522c55605d06/api/v1beta1/azurecluster_default.go#L160
+	// CAPZ enables NAT Gateways by default, so we are using this hack to disable
+	// nat gateways when we prefer to use load balancers for node egress.
+	nodeSubnetID := ""
+	if installConfig.Config.Platform.Azure.OutboundType != azure.NatGatewayOutboundType {
+		// Because the node subnet does not already exist, we are using an arbitrary value.
+		// We could populate this with the proper subnet ID in the case of BYO VNET, but
+		// the value currently has no practical effect.
+		nodeSubnetID = clusterID.InfraID
```

Comment:
It probably makes more sense to set this as a more explicitly unknown value such as `UNKNOWN`

Reply:
Addressed in https://github.com/openshift/installer/pull/8727/commits/7f8c6768edaf5bde9c45b498d326ab3799bf165e. If we like this, we can use tide to squash the new commit into a single one.